### ISSUE TITLE
Don't follow "Did you mean '...'?" redirection

### ISSUE THREAD

**Issue by fin0** [imankulov/linguee-api#23]
I used linguee-api to search for some translations from English to German and experienced the following problem:
If a word is rare and there is an alternative which is syntactically close and much more frequently used in the language, Linguee offers a "Did you mean '...'?" link at the top of the search results. If this is the case, linguee-api gives the result of the redirection and not the one of the searched word.
Examples (given by the URL linguee-api is querying):
https://www.linguee.com/english-german/search?query=wit&ajax=1
https://www.linguee.com/english-german/search?query=nod&ajax=1

I guess this is not by intention? I don't know, if this behavior occurs on all cases where the result is prefixed with a "Did you mean '...'?" question.

Thanks in advance :)

**Comment by imankulov** [comment]
Oh, thanks for reporting. Great catch!

I implemented automatic following for redirects, assuming that it would only fire on words without translations (mostly typos.) The "wit" case provides a counter-example --- a page having both translations and a suggestion.

It would probably make sense to introduce a new "follow_corrections" option having three variants.

- `follow_corrections=always`: Unconditionally follow a correction when finding one. Use it as the default option for backward compatibility.
- `follow_corrections=never`: Ignore corrections, and try to parse current text.
- `follow_corrections=on_empty_translations`: Parse the result and return if it has translations. If it has no translations, see if there's a correction. If correction is found, follow it.

Does it make sense, @fin0? Please let me know if you have any questions, concerns, or suggestions before I start implementing the `follow_corrections` flag. Thanks!

**Comment by imankulov** [comment]
Alright, the flag is implemented, as described above. The changes are live on the testing server.

[always follow corrections](https://linguee-api-v2.herokuapp.com/api/v2/translations?query=wit&src=en&dst=de&guess_direction=false&follow_corrections=always) (default and backward-compatible behavior)

![before](https://user-images.githubusercontent.com/75075/182224388-86fa198b-8e13-4d89-8077-e5e2c4ffe7cf.png)

[follow only on empty translations](https://linguee-api-v2.herokuapp.com/api/v2/translations?query=wit&src=en&dst=de&guess_direction=false&follow_corrections=on_empty_translations)

![after](https://user-images.githubusercontent.com/75075/182224376-d22e5562-603d-4d87-a3e3-cd51180c865d.png)

**Comment by fin0** [comment]
Hello @imankulov, wow that was fast! Thanks a lot for your quick response and implementation. Will do further testing once it is merged into master, but everything looks perfect already :)

**Comment by imankulov** [comment]
Hey @fin0, for the record, it's already merged with the master branch.
